cmd/config.go

```diff
@@ -151,50 +134,129 @@ func NewConfigSys() *ConfigSys {
 	return &ConfigSys{}
 }
 
+// Create config.json at the backend (.minio.sys/config/config.json or on etcd)
+// If config.json is not found at the backend, but found in old-path (~/.minio/config.json)
+// then move config.json from old-path to backend.
+func configInitBackend(objAPI ObjectLayer) error {
+	transactionConfigFile := path.Join(minioConfigPrefix, minioConfigFile) + ".transaction"
+
+	// As object layer's GetObject() and PutObject() take respective lock on minioMetaBucket
+	// and configFile, take a transaction lock to avoid data race between readConfig()
+	// and saveConfig().
+	objLock := globalNSMutex.NewNSLock(minioMetaBucket, transactionConfigFile)
+	if err := objLock.GetLock(globalOperationTimeout); err != nil {
+		return err
+	}
+	defer objLock.Unlock()
+
+	b, err := readServerConfig(context.Background(), objAPI)
+	if err != nil && err != errConfigNotFound {
+		return err
+	}
+	if err == nil {
+		// File already available, nothing more to be done.
+		return nil
+	}
+
+	// Construct path to config.json for the given bucket.
+	localConfigpath := getConfigFile()
+	if isFile(localConfigpath) {
+		b, err = ioutil.ReadFile(localConfigpath)
+		if err != nil {
+			return err
+		}
+		if err = saveServerConfig(context.Background(), objAPI, b); err != nil {
+			return err
+		}
+	} else {
+		// Create config.json of latest version
+		srvCfg := newServerConfig()
+
+		// Override any values from ENVs.
+		srvCfg.loadFromEnvs()
+
```

Comment:
No, note that in this part of the code we are only migrating.
loadConfig() happens at a later stage which does loadToCachedConfigs()

Reply:
that's not how it's happening though without loadToCachedConfigs() we are not honoring the JWKS and OPA urls here. loadConfig() indeed does that but its not sufficient because on Gateway that's not happening.

Reply:
looks like I am missing something which I am not able to see, we can discuss in person.